src/achievements.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include "config.hpp"
+
+struct achievement
+{
+	std::string id_;
+	std::string name_;
+	std::string name_completed_;
+	std::string description_;
+	std::string description_completed_;
+	std::string icon_;
+	std::string icon_completed_;
+	bool hidden_;
+	bool achieved_;
+
+	achievement(const config& cfg, bool achieved)
+		: id_(cfg["id"].str())
+		, name_(cfg["name"].str())
+		, name_completed_(cfg["name_completed"].str(cfg["name"].str()))
+		, description_(cfg["description"].str())
+		, description_completed_(cfg["description_completed"].str(cfg["description"].str()))
+		, icon_(cfg["icon"].str())
+		, icon_completed_(cfg["icon_completed"].str())
+		, hidden_(cfg["hidden"].to_bool())
+		, achieved_(achieved)
+	{
+	}
+};
+
+struct achievement_group
+{
+	std::string display_name_;
+	std::string content_for_;
+	std::vector<achievement> achievements_;
+
+	achievement_group(const config& cfg);
+};
+
+class achievements
```

Comment:
A bit of documentation would be good. From the names, I'd assume `achievement_group` and `achievements` would be the same thing.

Reply:
Added.